src/logger/DebugLogger.ts

```diff
@@ -6,93 +6,101 @@ import {PlatformTools} from "../platform/PlatformTools";
  * Performs logging of the events in TypeORM via debug library.
  */
 export class DebugLogger implements Logger {
-    private debug = PlatformTools.load("debug");
+    private debugPromise = import("debug");
+
+    private debugQueryLogPromise = this.debugPromise.then(debug => debug("typeorm:query:log"));
```

Comment:
Got it. Why not to use just `require` here btw?

Reply:
@alfaproject since typeorm is cross-platform we can't have dependencies to be imported directly, we need to use PlatformTools.load instead.

Reply:
This is the right way to do cross-platform though: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports

Reply:
i know what `import` is supposed to, but for now we need to use `PlatformTools.load` because of internal processes (for example we don't want for `fs` to be imported in browser AT ALL)

Reply:
I see. Is there a way I can see those processes? It sounds weird that this would be imported if it's not being used. That's the whole point of having dynamic imports. They are only imported if the code is used.

I'll revert btw, but just wondering.

Reply:
checkout BrowserPlatformTools and its usages that replaces original PlatformTools for browser. Point of PlatformTools.load isn't in dynamic loading, its point is in having some single point where external packages are being loaded. And this single place is being replaced by BrowserPlatformTools implementation for browser platform.